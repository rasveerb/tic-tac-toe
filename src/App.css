.container{
    text-align: center;
    background-color: #dec0f1;
    width: 100%;
    height: 100%;
    padding: 5% 0 50% 0;
  }
  
  .gameTitle{
    color: #a713f6;
    font-size: 40px;
    margin: 0;
    padding-right: 10%;
  }
  
  .playerTitle{
    color: #a713f6;
    font-size: 40px;
    margin: 0;
    padding-right: 8%;
  }
  
  .box{
    width: 80px;
    height: 80px;
    border: 4px solid #a713f6;
    display: inline-block;
    font-weight: 700;
    line-height: 80px;
    color: #a713f6;
  }
  
  .boardDisplay{
    display: flex;
    justify-content: center;
    align-items: center;
  }
  
  .gameBoard{
    display: inline-flex;
    justify-content: center;
    flex-wrap: wrap;
    width: 320px;
    color: #a713f6;
    padding-right: 10%;
  }
  
  .winnerBoard{
    display: inline-flex;
    justify-content: center;
    flex-wrap: wrap;
    width: 320px;
    color: #a713f6;
    padding-right: 0%;
  }
  
  .playAgain{
    color: #dc0073;
    font-weight: 700;
    border-radius: 12px;
    border: 2px solid #a713f6;
    height: 100px;
    width: 100px;
    background-color: #edabce;
  }
  
  .playAgain:disabled{
    display: none;
  }