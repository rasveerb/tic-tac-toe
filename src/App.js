import React, {Component} from 'react';
import './App.css';
import PlayerStatus from './components/PlayerStatus/PlayerStatus';


class App extends Component {
  constructor(props){
    super(props)
    this.state={
      board : Array(9).fill(null),
      player: null,
      winner: null,
    }
  }


  checkWinner(){
    let winningCombos = [
      //rows
      ["0","1","2"],
      ["3","4","5"],
      ["6","7","8"],

      //columns
      ["0","3","6"],
      ["1","4","7"],
      ["2","5","8"],

      //diagnols
      ["0","4","8"],
      ["2","4","6"]
    ]

    for(let index = 0; index < winningCombos.length; index++){
      const [a, b, c] = winningCombos[index];

      if(this.state.board[a] && this.state.board[a] === this.state.board[b] && this.state.board[a] === this.state.board[c]){
        this.setState({
          winner: this.state.player,
        });

      }
    }
  }

  handleClick(index){
    if(this.state.player && !this.state.winner) {

      let newBoard = this.state.board;

      if(this.state.board[index] === null ) {
        newBoard[index] = this.state.player

        this.setState({

          board: newBoard,

          player: this.state.player === "X" ? "O" : "X"
        })

        this.checkWinner()
      }
    }

  }

  setPlayer(player){
    this.setState({ player }) //instead of player:player - because assigning same thing
  }

  renderBoxes(){
    return this.state.board.map(

      (boxValue, index) =>

        <div className="box"

          key={index}

          onClick={(e) => this.handleClick(index)}>

          {boxValue}

        </div>
      )
  }

  resetGame(){
    this.setState({
      player: null,
      winner: null,
      board: Array(9).fill(null)
  })
}

  render(){
    return (

      <div className="container">

        <h1 className={!this.state.player ? "gameTitle" : "playerTitle"}> TIC TAC TOE </h1>

        <PlayerStatus player={this.state.player} setPlayer={(e)=>{this.setPlayer(e)}} winner ={this.state.winner}/>

        <div className = "boardDisplay">

            <div className={!this.state.winner ? "gameBoard" : "winnerBoard"}>

              {this.renderBoxes()}

            </div>

            <button className = "playAgain" disabled={!this.state.winner} onClick={() => this.resetGame()}> Play Again </button>

        </div>

      </div>
    );
  }
}

export default App;