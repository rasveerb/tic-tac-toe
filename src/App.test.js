import React from 'react';
import App from './App';

import {shallow, mount} from 'enzyme'

describe('App tests', () => {

    it('renders without crashing', () => {
      const wrapper = mount(<App />);

      expect(wrapper.debug()).toMatchSnapshot();
    });
    
//     it('renders player status correctly', () => {
//       const props = {
//         winner: 'X',
//         board : Array(9).fill(null),
//         player: null
//       }

//       const wrapper = mount(<App playerStatus={props.player}/>)

//       console.log("board: ", wrapper)

// const winner = wrapper.find('div.winnerBoard').text()
// expect(winner).toEqual('X')
//   });
});