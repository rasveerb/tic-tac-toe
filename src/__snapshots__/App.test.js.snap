// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests renders without crashing 1`] = `
"<App>
  <div className=\\"container\\">
    <h1 className=\\"gameTitle\\">
       TIC TAC TOE 
    </h1>
    <PlayerStatus player={{...}} setPlayer={[Function: setPlayer]} winner={{...}}>
      <div>
        <ChoosePlayer player={[Function: player]}>
          <form onSubmit={[Function: onSubmit]}>
            <label className=\\"playerX\\">
              <div className=\\"inputDisplay\\">
                <div className=\\"playerText\\">
                  Player X
                </div>
                <input className=\\"inputX\\" type=\\"radio\\" name=\\"player\\" value=\\"X\\" onChange={[Function: onChange]} />
              </div>
            </label>
            <label className=\\"playerO\\">
              <div className=\\"inputDisplay\\">
                <div className=\\"playerText\\">
                  Player O
                </div>
                <input className=\\"inputO\\" type=\\"radio\\" name=\\"player\\" value=\\"O\\" onChange={[Function: onChange]} />
              </div>
            </label>
            <button className=\\"startButton\\" type=\\"submit\\" disabled={true}>
               Start
            </button>
          </form>
        </ChoosePlayer>
      </div>
    </PlayerStatus>
    <div className=\\"boardDisplay\\">
      <div className=\\"gameBoard\\">
        <div className=\\"box\\" onClick={[Function: onClick]} />
        <div className=\\"box\\" onClick={[Function: onClick]} />
        <div className=\\"box\\" onClick={[Function: onClick]} />
        <div className=\\"box\\" onClick={[Function: onClick]} />
        <div className=\\"box\\" onClick={[Function: onClick]} />
        <div className=\\"box\\" onClick={[Function: onClick]} />
        <div className=\\"box\\" onClick={[Function: onClick]} />
        <div className=\\"box\\" onClick={[Function: onClick]} />
        <div className=\\"box\\" onClick={[Function: onClick]} />
      </div>
      <button className=\\"playAgain\\" disabled={true} onClick={[Function: onClick]}>
         Play Again 
      </button>
    </div>
  </div>
</App>"
`;
